lib/utils/typeGuards.mjs

```diff
@@ -49,6 +49,14 @@ export function isDocument(node) {
 	return node.type === 'document';
 }
 
+/**
+ * @param {import('postcss').Node} node
+ * @returns {boolean}
+ */
+export function isInDocument({ parent }) {
+	return Boolean(parent && (isDocument(parent) || 'document' in parent));
+}
```

Comment:
This is not a type guard.

Maybe better to keep this function local to the rule?